src/VisualStudio/Core/Def/LanguageService/AbstractCreateServicesOnTextViewConnection.cs

```diff
@@ -56,9 +57,12 @@ public AbstractCreateServicesOnTextViewConnection(
                 listenerProvider.GetListener(FeatureAttribute.CompletionSet),
                 threadingContext.DisposalToken);
 
-        Workspace.DocumentOpened += QueueWorkOnDocumentOpened;
+        _workspaceDocumentOpenedDisposer = Workspace.RegisterDocumentOpenedHandler(QueueWorkOnDocumentOpened);
     }
 
+    public void Dispose()
+        => _workspaceDocumentOpenedDisposer.Dispose();
```

Comment:
i guess this is better than leaking.  i'm not actually validating this connection type actually gets disposed itself properly.

Reply:
The derived implementations are mef exports, so if their catalog gets disposed, this will now get disposed. I can remove if you prefer just not to store/use the disposer.

Reply:
no.  i prefer this.  always oogy when any sort of event is registered and not unregistered.  if this breaks, then we can figure out why and fix later.